core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -428,11 +432,12 @@ impl PrioGraphScheduler {
             .map(|thread_id| {
                 (
                     thread_id,
+                    batch_cus_per_thread[thread_id] + in_flight_cus_per_thread[thread_id],
                     batches_per_thread[thread_id].len() + in_flight_per_thread[thread_id],
                 )
             })
-            .min_by(|a, b| a.1.cmp(&b.1))
-            .map(|(thread_id, _)| thread_id)
+            .min_by(|a, b| a.1.cmp(&b.1).then_with(|| a.2.cmp(&b.2)))
```

Comment:
Still use the number of txs as a backup comparison if the CUs happen to be exactly equal